pytorch_lightning/utilities/auto_restart.py

```diff
@@ -545,3 +546,37 @@ def _add_capture_metadata_collate(dataloader: DataLoader) -> None:
     dataloader.collate_fn = partial(
         _capture_metadata_collate, dataset=dataloader.dataset, default_collate=dataloader.collate_fn
     )
+
+
+def _reload_state_dict(dataloader: DataLoader, state_dict: Dict[str, Any]) -> None:
```

Comment:
```suggestion
def reload_dataloader_state_dict(dataloader: DataLoader, state_dict: Dict[str, Any]) -> None:
```

* add `dataloader` infix to make clear what statedict this method loads
* remove `_` prefix since you don't use it in this file only but import it in the other one -> therefore it should not be protected

Reply:
Makes sense :)